src/components/ConfigEditor/ConfigEditor.test.tsx

```diff
@@ -44,4 +44,47 @@ describe('ConfigEditor', () => {
 
     await waitFor(() => expect(refreshSchemaSpy).toHaveBeenCalledTimes(2));
   });
+
+  it('should shows the beta OBO toggle if feature gate enabled', async () => {
+    const originalConfigValue = grafanaRuntime.config.featureToggles.adxOnBehalfOf;
+
+    grafanaRuntime.config.featureToggles.adxOnBehalfOf = true;
+
+    render(<ConfigEditor {...mockConfigEditorProps()} />);
+
+    await waitFor(() => expect(screen.getByLabelText('Use On-Behalf-Of')).toBeInTheDocument());
+
+    // reset config to not impact future tests
+    grafanaRuntime.config.featureToggles.adxOnBehalfOf = originalConfigValue;
```

Comment:
this needs to be on a `beforeEach` hook: if the test fails, this line is not executed.

Reply:
oops indeed! 🤭  Copy pasted without thinking.